src/CliMenu.php

```diff
@@ -272,25 +275,57 @@ private function display() : void
     protected function moveSelection(string $direction) : void
     {
         do {
-            $itemKeys = array_keys($this->items);
-
-            $direction === 'UP'
-                ? $this->selectedItem--
-                : $this->selectedItem++;
-
-            if (!array_key_exists($this->selectedItem, $this->items)) {
-                $this->selectedItem  = $direction === 'UP'
-                    ? end($itemKeys)
-                    : reset($itemKeys);
-            } elseif ($this->getSelectedItem()->canSelect()) {
-                return;
+            if ($direction === 'UP' || $direction === 'DOWN') {
+                $itemKeys = array_keys($this->items);
+
+                $direction === 'UP'
+                    ? $this->selectedItem--
+                    : $this->selectedItem++;
+
+                if (!array_key_exists($this->selectedItem, $this->items)) {
+                    $this->selectedItem  = $direction === 'UP'
+                        ? end($itemKeys)
+                        : reset($itemKeys);
+                } elseif ($this->getSelectedItem()->canSelect()) {
+                    return;
+                }
+            } else {
+                $item = $this->getSelectedItem(true);
+                if (!$item instanceof SplitItem) {
+                    return;
+                }
+
+                $itemKeys = array_keys($item->getItems());
+                $selectedItemIndex = $item->getSelectedItemIndex();
+                $direction === 'LEFT'
+                    ? $selectedItemIndex--
+                    : $selectedItemIndex++;
+                $item->setSelectedItemIndex($selectedItemIndex);
+
+                if (!array_key_exists($selectedItemIndex, $item->getItems())) {
+                    $selectedItemIndex = $direction === 'LEFT'
+                        ? end($itemKeys)
+                        : reset($itemKeys);
+                    $item->setSelectedItemIndex($selectedItemIndex);
+                } elseif ($item->getItems()[$item->getSelectedItemIndex()]->canSelect()) {
+                    return;
+                }
             }
         } while (!$this->getSelectedItem()->canSelect());
     }
 
-    public function getSelectedItem() : MenuItemInterface
+    public function getSelectedItem(bool $oneLevelDeep = false) : MenuItemInterface
     {
-        return $this->items[$this->selectedItem];
+        if ($oneLevelDeep) {
```

Comment:
I made it so that `getSelectedItem` returns the right element.
It means that if we are selecting an item (let's say a **SelectableItem**) which is inside a **SplitItem**, it will return the **SelectableItem** and not the **SplitItem**.

But what if we want the "first level" selected element (so **SplitItem** in our example) ? Then we set the argument to `true` and the function will not "dig" layers to find the right one.

_If we one day allow split items inside split items or something else, then it would still work as intended._
_If we add some new element that allows multiple item inside itself then it would still do its job too._

**The point is not to check if the element is a SplitItem (event though it's now used like that), but to find what is the top level element that is selected.**

I agree that it is currently only used when trying to move the selection left or right, but since it's a public function it can also be used outside of the class so that (for example) a keyboard shortcut could have different effects depending on the element currently selected.